framework/wazuh/utils.py

```diff
@@ -741,8 +741,9 @@ class WazuhDBBackend(AbstractDatabaseBackend):
     """
     This class describes a wazuh db backend that executes database queries
     """
-    def __init__(self, agent_id):
-        self.agent_id = agent_id
+    def __init__(self, agent_id=None, mitre=False):
```

Comment:
I suggest changing `mitre` param by another more generic like `query_format` for example. This new param should accept a string instead of a boolean. The string could be used to match a private method to render the proper message:
_render_{query_format}, where `query_format` could be `agent` or `mitre`. There would be two private methods:
- `_render_agent` which returns a query in the form: agent {self.agent_id} sql {query}
- `_render_mitre` which returns a query in the form: mitre sql {query}